pkg/controller/volume/persistentvolume/pv_controller_base.go

```diff
@@ -94,7 +96,8 @@ func NewController(p ControllerParameters) (*PersistentVolumeController, error)
 		volumeQueue:                   workqueue.NewNamed("volumes"),
 		resyncPeriod:                  p.SyncPeriod,
 		operationTimestamps:           metrics.NewOperationStartTimeCache(),
-		translator:                    csitrans.New(),
+		translator:                    p.CSITranslator,
+		csiMigratedPluginManager:      p.CSIMigratedPluginManager,
```

Comment:
I don't think its that weird to have both of these (especially with the modifications I suggest in the other comments). One is the actual translator - the other is a plugin manager that happens to also depend on some `InTreePluginNameGetter` interface which just happens to be a part of the `CSITranslator` struct.

The thing is I've noticed `InTreePluginNameGetter` seems to stand on its own an is used by itself in multiple places so maybe it would be a good idea to split that out into a different package inside the translation library. However, I don't think its that necessary and may just increase complexity.